### PR TITLE
Prompt user to verify account when they attempt to publish a post

### LINKED ISSUE #4600: NUX: Add email confirmation prompt for WordPress.com users
### Expected behavior

When a WordPress.com user signs up on the web, they are prompted to confirm their email address at the end of the signup flow:

screen shot 2016-10-04 at 11 29 36 am

If the user doesn't confirm their email address at that step, they are blocked from publishing new posts and see a notice/reminder in the Calypso editor:

screen shot 2016-10-04 at 11 32 14 am

screen shot 2016-10-04 at 11 33 13 am

Currently, those same users are blocked from publishing in the WordPress app as well, but the publish button isn't deactivated — their posts just inexplicably become drafts instead of publishing. (They often have to reach out to support for an explanation of what's happening and for help confirming their email address.)

Instead, we should offer a similar notice/reminder within the app to help these users.
### Steps to reproduce the behavior
1. While logged out of WordPress.com, open https://wordpress.com/start
2. Sign up for a new account but don't confirm your email address
3. In the app, log in to your new account
4. Create a new post and try to publish it

Result: The post becomes a draft.
##### Tested on Nexus 9, Android 7.0, WPAndroid alpha-22

Related issue for iOS: https://github.com/wordpress-mobile/WordPress-iOS/issues/6035

### PR DESCRIPTION
**Pull request by tonyr59h**
Fixes #4600 by adding a dialog to prompt the user to verify their account. The prompt is presented when the user selects `Publish` from the post editor and their Account info indicates that they haven't verified their account (via email).

**NOTE** - ~~DO NOT MERGE until wordpress-mobile/WordPress-FluxC-Android#379 is merged. There's a commit hash included (in _this_ PR) that needs to be reverted after that merge.~~

To test:
* Create a new account but DO NOT click the link in the verification email
* Log into the new account and create a new post
* Attempt to publish the post

cc @rachelmcr, would you mind giving this a whirl? I'd appreciate your feedback on the UX here, I think we can make things more clear but I'm not sure exactly how.

### CODE REVIEW

**Comment by tonyr59h** [comment]
The FluxC PR has been merged! Time to give this branch some attention. I'll merge `develop` in and run through some testing again. @rachelmcr have you had a chance to test this one out?

**Comment by tonyr59h** [comment]
Merged and happy-path tested, everything looks good.

**Comment by rachelmcr** [comment]
I just tested this, and it feels ok to me. I like the "resend email" option and that tapping "ok" saves my post as a draft. Would it be possible to include the user's email on this popup? That additional information feels helpful, e.g. if I want to check my email address before resending the confirmation email.

The only other thing is that the message feels a little out of nowhere if I'm expecting to publish my post:



On the web, there's an initial message (before I try to publish) with this language:

> To publish, check your email and confirm your address. Learn More

I don't think we should try to put that message in the editor itself (to much potential for clutter, especially on small screens) but maybe we could incorporate it into this popup to add a little more context. Something like:

> Please confirm your email address
> We sent you an email when you first signed up. Please open the message and click the blue button to enable publishing.

In any case, this is an immense improvement over the current behavior.

One last thought: For a future iteration we might consider what's being discussed on the iOS side in https://github.com/wordpress-mobile/WordPress-iOS/issues/6035#issuecomment-286297879 (making the email confirmation part of the app login process rather than putting the notice in the editor).

**Comment by tonyr59h** [comment]
Thanks for the awesome feedback!

> ...potential for clutter

Agreed, but I see the need for more information here. I'll improve the strings and see about adding the email address.

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1665,4 +1665,14 @@
...
     <string name="photo_picker_title">Choose photo</string>
 
+    <!-- E-mail verification reminder dialog -->
+    <string name="toast_saving_post_as_draft">Saving post as draft</string>
+    <string name="toast_verification_email_sent">Verification email sent, check your inbox</string>
+    <string name="toast_verification_email_send_error">Error sending verification email. Are you already verified?</string>
+    <string name="editor_confirm_email_prompt_title">Please confirm your email address</string>
+    <string name="editor_confirm_email_prompt_message">We sent you an email when you first signed up. Please open the message and click the blue button to enable publishing.</string>
+    <string name="editor_confirm_email_prompt_message_with_email">We sent an email to %s when you first signed up. Please open the message and click the blue button to enable publishing.</string>
+    <string name="editor_confirm_email_prompt_positive" translatable="false">@string/ok</string>
```
Comment (maxme) [inline WordPress/src/main/res/values/strings.xml]: Same as the other PR, we should drop this and use `android.R.string.ok` directly. Also I guess we could drop the "ok" string from here and use `android.R.string.ok` everywhere.

**Comment by tonyr59h** [comment]
I switched to using `android.R.string.ok` for this PR. I've got a stash with our `ok` string resource removed and usages replaced that I figured another PR would be more appropriate for.

**Comment by maxme** [comment]
Looks good to me. I agree the UI is not super friendly. We could eventually disable the publish button and show a tooltip message (like the hint we show for updating the Avatar on the Me tab), but that feels like overkill.

**Pull request merged**